cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/JiraBucketWidget.kt

```diff
@@ -0,0 +1,58 @@
+package com.cognifide.cogboard.widget.type
+
+import com.cognifide.cogboard.widget.AsyncWidget
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import com.cognifide.cogboard.CogboardConstants as CC
+
+class JiraBucketWidget(vertx: Vertx, config: JsonObject) : AsyncWidget(vertx, config) {
+
+    private val bucketQueries: JsonArray = config.getJsonArray("bucketQueries")
+
+    private val buckets: JsonArray = prepareBuckets()
+
+    private fun prepareBuckets(): JsonArray {
+        val preparedBuckets = JsonArray()
+        bucketQueries.forEach {
+            if (it is JsonObject) {
+                val bucket = JsonObject()
+                bucket.put("id", it.getString("id"))
+                bucket.put("name", it.getString("bucketName"))
+                preparedBuckets.add(bucket)
+            }
+        }
+        return preparedBuckets
+    }
+
+    override fun handleResponse(responseBody: JsonObject) {
+        val response = JsonObject()
+        val issues = responseBody.getJsonArray("issues").size()
+        val bucketId = responseBody.getString("requestId")
+
+        buckets.forEach {
+            val key = (it as JsonObject).getString("id")
+            if (bucketId == key) {
+                it.put("issueCounts", issues)
```

Comment:
either `brake` after finding or reimplement with stream and filter

Reply:
Reimplemented

`(buckets.first { compareId(bucketId, it) } as JsonObject)
                .put("issueCounts", issues)`

`private fun compareId(bucketId: String?, it: Any?) = bucketId == (it as JsonObject).getString(CC.PROP_ID)`

Reply:
where?

Reply:
in 5e9823c389f17db7ef25c5707c73438bb2f11956